osu.Game/Database/OsuDbContext.cs

```diff
@@ -58,11 +58,20 @@ public OsuDbContext(string connectionString)
             this.connectionString = connectionString;
 
             var connection = Database.GetDbConnection();
-            connection.Open();
-            using (var cmd = connection.CreateCommand())
+            try
+            {
+                connection.Open();
+
+                using (var cmd = connection.CreateCommand())
+                {
+                    cmd.CommandText = "PRAGMA journal_mode=WAL;";
+                    cmd.ExecuteNonQuery();
+                }
+            }
+            catch (Exception e)
             {
-                cmd.CommandText = "PRAGMA journal_mode=WAL;";
-                cmd.ExecuteNonQuery();
+                connection.Close();
+                throw;
```

Comment:
`using (var connection = Database.GetDbConnection()) { }` will handle the `.Close()` for you, and prevent a memory leak if the command doesn't fail.

Reply:
not closing the connection is intentional here as the context continue to use it (and manage its lifetime)